### DISCUSSION TITLE
Selecting Multiple Nodes

### DISCUSSION THREAD

**Discussion by mjcee** [xyflow/xyflow#4765] [Q&A]
I am using the ReactFlow Pro AutoLayout as the basis for my application.  When the leva panel is displayed in the upper right corner, I am able to select multiple nodes using the <ctrl> key to select the second node.  When I comment out the leva panel and set my LayoutOptions directly, I am no longer able to select multiple nodes.

You can duplicate this problem by:

 Comment out line 138 - onNodeClick={onNodeClick} to stop the on click from creating a new node and allowing the click to select the node.

 Comment out the leva panel stating at line 54 and ending at line 82.  

 Insert this line after line 82 to set the LayoutOptions

    useAutoLayout({ algorithm: 'elk', direction: 'TB', spacing: [50, 50] } as LayoutOptions); 

You now cannot select multiple nodes.

You can change the following lines in index.css to better see the node outline when a node is selected:

  --background-color: white;
  --node-color: #0f172a;

Since I do not want the leva panel displayed in my app, how can I select multiple nodes without the leva panel?

**Comment by chrtze** [comment]
Hey @mjcee, thanks for pointing this out. Here is why it happens and how to fix this:

When you change the hook call to `useAutoLayout({ algorithm: 'elk', direction: 'TB', spacing: [50, 50] } as LayoutOptions);` a new object with the layout options will be created on every render, causing the layout algorithm to run indefinitely. This is because `options` is a dependency of the `useEffect` call in `useAutoLayout`.

To fix this, you can define the layout options object outside of your component like this:

```tsx
const layoutOptions = {
  algorithm: 'elk',
  direction: 'TB',
  spacing: [50, 50],
} as LayoutOptions;
```

Then, use it in your component like this:

```tsx
  useAutoLayout(layoutOptions);
```

Hope that this fixes your problem. We'll raise this issue internally and see if we can improve the example.
